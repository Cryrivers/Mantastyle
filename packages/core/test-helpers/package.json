{
  "name": "@manta-style/test-helpers",
  "version": "0.2.0-alpha.9",
  "main": "src/index.ts",
  "dependencies": {
    "prettier": "^1.14.3"
  },
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "@types/prettier": "^1.13.2"
  }
}
