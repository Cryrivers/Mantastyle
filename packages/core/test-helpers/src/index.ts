import * as BuilderPluginTestHelper from './builder-tester';
export { BuilderPluginTestHelper };
