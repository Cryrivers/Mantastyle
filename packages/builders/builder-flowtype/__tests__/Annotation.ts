import getTranspiledString from '../src/utils/transpile';
import { BuilderPluginTestHelper } from '@manta-style/test-helpers';

const testTranspiledString = BuilderPluginTestHelper.testTranspiledString(
  getTranspiledString,
);

describe('Annotation', () => {
  test('jsdoc parser', () => {
    expect(
      testTranspiledString(`
    /**
     * @mantastyle {{ asdf asdf asdf }}
     * 
     * @example 12345 67890
     * @iterate 23456 34567
     *
     */
    type Test = {};
    `),
    ).toMatchSnapshot();
  });
});
