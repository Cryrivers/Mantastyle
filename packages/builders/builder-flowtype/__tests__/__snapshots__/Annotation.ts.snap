// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Annotation jsdoc parser 1`] = `
"FROM:

    /**
     * @mantastyle {{ asdf asdf asdf }}
     * 
     * @example 12345 67890
     * @iterate 23456 34567
     *
     */
    type Test = {};
    

--------------------------------------------------------
TO:
'use strict';

var _flowtypeHelpers = _interopRequireDefault(
  require('@manta-style/flowtype-helpers'),
);

var _runtime = _interopRequireDefault(require('@manta-style/runtime'));

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj };
}

/**
 * @mantastyle {{ asdf asdf asdf }}
 *
 * @example 12345 67890
 * @iterate 23456 34567
 *
 */
var Test = _runtime.default.TypeAliasDeclaration(
  'Test',
  function(typeFactory) {
    var type = _runtime.default.TypeLiteral(function(typeLiteral) {});

    return type;
  },
  [
    {
      key: '@mantastyle',
      value: '{{ asdf asdf asdf }}',
    },
    {
      key: '@example',
      value: '12345 67890',
    },
    {
      key: '@iterate',
      value: '23456 34567',
    },
  ],
);
"
`;
